modGui/src/org/aion/gui/controller/HeaderPaneControls.java

```diff
@@ -0,0 +1,150 @@
+import org.aion.log.LogEnum;
+import org.slf4j.Logger;
+
+import java.awt.*;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+public class HeaderPaneControls extends AbstractController {
+
+    private static final Logger log = AionLoggerFactory.getLogger(LogEnum.GUI.name());
+
+    private static final String AION_URL = "http://www.aion.network";
+
+    private static final String STYLE_DEFAULT = "default";
+
+    private static final String STYLE_PRESSED = "pressed";
+
+//    private final BlockchainConnector blockchainConnector = BlockchainConnector.getInstance();
+
+    private final Map<Node, HeaderPaneButtonEvent> headerButtons = new HashMap<>();
+
+    @FXML
+    private TextField accountBalance;
+    @FXML
+    private TextField activeAccount;
+    @FXML
+    private Label activeAccountLabel;
+    @FXML
+    private VBox homeButton;
+    @FXML
+    private VBox settingsButton;
+
+    private String accountAddress;
+
+    @Override
+    public void internalInit(URL location, ResourceBundle resources) {
+        headerButtons.put(homeButton, new HeaderPaneButtonEvent(HeaderPaneButtonEvent.Type.OVERVIEW));
+        headerButtons.put(settingsButton, new HeaderPaneButtonEvent(HeaderPaneButtonEvent.Type.SETTINGS));
+
+        clickButton(homeButton);
+    }
+
+    @Override
+    protected void registerEventBusConsumer() {
+        super.registerEventBusConsumer();
+        EventBusRegistry.INSTANCE.getBus(EventPublisher.ACCOUNT_CHANGE_EVENT_ID).register(this);
+    }
+
+    public void openAionWebSite() {
+        try {
+            Desktop.getDesktop().browse(new URI(AION_URL));
+        } catch (IOException | URISyntaxException e) {
+            log.error("Exception occurred trying to open website: %s", e.getMessage(), e);
+        }
+    }
+
+    public void handleButtonPressed(final MouseEvent pressed) {
+        for (final Node headerButton : headerButtons.keySet()) {
+            ObservableList<String> styleClass = headerButton.getStyleClass();
+            styleClass.clear();
+            if (pressed.getSource().equals(headerButton)) {
+                styleClass.add(STYLE_PRESSED);
+                setStyleToChildren(headerButton, "header-button-label-pressed");
+                HeaderPaneButtonEvent headerPaneButtonEvent = headerButtons.get(headerButton);
+                sendPressedEvent(headerPaneButtonEvent);
+            } else {
+                styleClass.add(STYLE_DEFAULT);
+                setStyleToChildren(headerButton, "header-button-label");
+            }
+        }
+    }
+
+    private void setStyleToChildren(Node headerButton, String styleClassToSet) {
+        VBox vbox = (VBox) ((VBox) headerButton).getChildren().get(0);
+        ObservableList<String> styleClass = vbox.getChildren().get(0).getStyleClass();
```

Comment:
In this case yes, because the headerButton parameter is intended to be one of the buttons in the top navigation pane of the GUI.  The buttons are always a VBox (a rectangular box) and then has a Label as child (the text of the button).  Nothing prevents someone from using that setStyleToChildren method wrong though -- they could call with the headerButton parameter being something else, possibly with 0 children.